src/kernels/variables/pythonVariableRequester.ts

```diff
@@ -36,6 +37,24 @@ export function parseDataFrame(df: DataFrameSplitFormat) {
     return { data };
 }
 
+export async function safeExecuteSilently(
+    kernel: IKernel,
+    code: string,
+    errorOptions?: SilentExecutionErrorOptions
+): Promise<nbformat.IOutput[]> {
+    if (kernel.disposed || kernel.disposing || !kernel.session || !kernel.session.kernel || kernel.session.disposed) {
```

Comment:
If the kernel has died/dying, then no point attempting to send requests for variables.